src/services/business/fetch-business-details-service.js

```diff
@@ -1,39 +1,34 @@
 /**
- * Fetches the business details associated with the logged in users business
+ * Returns the `businessDetails` from the session cache if defined,
+ * otherwise queries the DAL, maps the response, updates the cache
+ * and returns the mapped payload
+ *
  * @module fetchBusinessDetailsService
  */
 
+import { dalConnector } from '../../dal/connector.js'
+import { businessDetailsQuery } from '../../dal/queries/business-details.js'
+import { mapBusinessDetails } from '../../mappers/business-details-mapper.js'
+
 const fetchBusinessDetailsService = async (yar) => {
-  // Refactor: Remove stubbed data and instead call the API to get the business details associated with the users log in
-  // This will be using the consolidated view API
-  // The data needed for the business details page.
-
-  const businessDetails = yar.get('businessDetails')
-    ? yar.get('businessDetails')
-    : {
-        businessName: 'Agile Farm Ltd',
-        businessAddress: {
-          address1: '10 Skirbeck Way',
-          address2: '',
-          city: 'Maidstone',
-          county: '',
-          postcode: 'SK22 1DL',
-          country: 'United Kingdom'
-        },
-        businessTelephone: '01234567890',
-        businessMobile: '01234567890',
-        businessEmail: 'a.farmer@farms.com',
-        sbi: '123456789',
-        vatNumber: '',
-        tradeNumber: '987654',
-        vendorRegistrationNumber: '699368',
-        countyParishHoldingNumber: '12/563/0998',
-        businessLegalStatus: 'Sole proprietorship',
-        businessType: 'Central or local government',
-        userName: 'Alfred Waldron'
-      }
-  yar.set('businessDetails', businessDetails)
-  return businessDetails
+  return yar.get('businessDetails') ?? await getFromDal(yar)
+}
+
+const getFromDal = async (yar) => {
+  // replace variables and email when defraId is setup
+  const variables = { sbi: '107183280', crn: '9477368292' }
+  const email = 'not-a-real-email@test.co.uk'
+
+  const dalResponse = await dalConnector(businessDetailsQuery, variables, email)
+
+  if (dalResponse.data) {
+    const mappedResponse = mapBusinessDetails(dalResponse.data)
+    yar.set('businessDetails', mappedResponse)
+
+    return mappedResponse
+  }
+
+  return dalResponse
```

Comment:
Consider throwing error here and create an error-handler function to handle the thrown error.  - I think we should create a new ticket for it.

Reply:
Agree - we need to look into how the DAL connector handles errors because the GQL response can contain errors and valid data.